astropy/tests/helper.py

```diff
@@ -409,11 +410,18 @@ def generic_recursive_equality_test(a, b, class_history):
     Check if the attributes of a and b are equal. Then,
     check if the attributes of the attributes are equal.
     """
-    dict_a = a.__getstate__() if hasattr(a, '__getstate__') else a.__dict__
+    if hasattr(a, '__getstate__'):
+        if not PYTHON_LT_3_11:
+            # NOTE: The call may need to be adapted if other objects implementing a __getstate__
+            # with required argument(s) are passed to this function.
+            dict_a = a.__getstate__(a) if inspect.isclass(a) else a.__getstate__()
+        else:
+            dict_a = a.__getstate__()
+    else:
+        dict_a = a.__dict__
```

Comment:
```suggestion
    if PYTHON_LT_3_11: 
        dict_a = a.__getstate__() if hasattr(a, '__getstate__') else a.__dict__
    else:
        # NOTE: The call may need to be adapted if other objects implementing a __getstate__
        # with required argument(s) are passed to this function.
        dict_a = a.__getstate__(a) if inspect.isclass(a) else a.__getstate__()
```

Reply:
Great. Thanks! I guess people will only ever pass object into this now that I see the rest of the logic in that function.